core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -296,7 +296,7 @@ private Object readResolve() {
     public void onAttached(Run<?, ?> r) {
         ParametersDefinitionProperty p = r.getParent().getProperty(ParametersDefinitionProperty.class);
         if (p != null) {
-            this.parameterDefinitionNames = p.getParameterDefinitionNames();
+            this.parameterDefinitionNames = new ArrayList<>(p.getParameterDefinitionNames());
```

Comment:
maybe unmodifyableList ? NIT

Reply:
No, that produces a weirder XStream form. Since it is `private`, there is no real safety advantage anyway.